src/Npgsql/NpgsqlActivitySource.cs

```diff
@@ -0,0 +1,51 @@
+﻿using Npgsql.Internal;
+using System;
+using System.Diagnostics;
+
+namespace Npgsql
+{
+    static class NpgsqlActivitySource
+    {
+#if NET5_0_OR_GREATER
+        static readonly ActivitySource Source = new("Npgsql");
+
+        public static bool IsEnabled => Source.HasListeners();
+
+        public static IDisposable? CommandStart(NpgsqlConnector connector, string sql)
+        {
+            var activity = Source.StartActivity(connector.Settings.Database!, ActivityKind.Client);
+            if (activity is not null)
+            {
+                activity.SetTag("db.system", "postgresql");
+                activity.SetTag("db.connection_string", connector.UserFacingConnectionString);
+                activity.SetTag("db.user", connector.Settings.Username);
+                activity.SetTag("db.name", connector.Settings.Database);
+                activity.SetTag("db.statement", sql);
+                activity.SetTag("db.connection.id", connector.Id);
```

Comment:
It should be pretty easy to add net.peer.ip, net.peer.name, net.peer.port, net.transport no? Note that either net.peer.ip or net.peer.name are considered mandatory in the [spec draft](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/database.md).